Zend/zend_compile.c

```diff
@@ -8978,6 +8978,10 @@ ZEND_API bool zend_binary_op_produces_error(uint32_t opcode, const zval *op1, co
 		/* Division by zero throws an error. */
 		return 1;
 	}
+	if ((opcode == ZEND_POW) && zval_get_long(op1) == 0 && zval_get_long(op2) < 0) {
```

Comment:
`<` should be `<=`

Reply:
I fixed the comment below. It applies to negative exponents, not 0.

Reply:
FWIW: It was not part of the RFC, but `0` raised to `0` would also be undefined.

Reply:
> FWIW: It was not part of the RFC, but 0 raised to 0 would also be undefined.

I also wondered about this. But apparently it is not universally considered undefined. See https://en.wikipedia.org/wiki/Zero_to_the_power_of_zero.

Reply:
According to [IEEE 754-2008](https://en.wikipedia.org/wiki/Zero_to_the_power_of_zero#IEEE_floating-point_standard)

> pow (x, ±0) is 1 for any x (even a zero, quiet NaN, or infinity)

There is no specialized `powr()` function in PHP which treats 0**0 as NaN.

Reply:
> According to IEEE 754-2008

For `fpow()` that absolutely makes sense. However for `pow()` and `**`, PHP could opt for a different behavior that would throw on the undefined operation. Like division by zero is also well-defined for `fdiv()`, but throws with `/`.

Reply:
That's true. Although it's not so unambiguous. We would need to create another RFC for this.

Reply:
Yes, that would not be covered this RFC. I primarily wanted to note it for posterity, because it is related.